nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -455,13 +409,11 @@ private static Type.ClassType typeWithPreservedAnnotations(
       TypeMetadata typeMetadata =
           new TypeMetadata(new TypeMetadata.Annotations(nullableAnnotationCompound));
       Type currentTypeArgType = castToNonNull(ASTHelpers.getType(curTypeArg));
-      if (currentTypeArgType.getTypeArguments().size() > 0) {
-        // nested generic type; recursively preserve its nullability type argument annotations
-        currentTypeArgType =
-            typeWithPreservedAnnotations((ParameterizedTypeTree) curTypeArg, state);
+      List<Type> genericArgType = currentTypeArgType.accept(TYPE_ARG_VISITOR, null);
+      if (genericArgType.size() > 0) {
+        currentTypeArgType = curTypeArg.accept(new PreservedAnnotationTreeVisitor(state), null);
       }
```

Comment:
the purpose of TYPE_ARG_VISITOR is just to be able to get the type arguments even if the Argument is not a ClassType, but the if condition checking the size is greater than 0 was already there as given below, do you want me to remove that logic? 

`if (currentTypeArgType.getTypeArguments().size() > 0) {
        // nested generic type; recursively preserve its nullability type argument annotations
        currentTypeArgType =
            typeWithPreservedAnnotations((ParameterizedTypeTree) curTypeArg, state);
      }`

Reply:
You mentioned you get an NPE without this check. What is the stack trace for that?

Reply:
One of the test cases that failed is: `A<A<@Nullable String>[]> var = new A<A<String>[]>();`, when we traverse the ParametrizedTypeTree on the RHS (A<A<String>[]>), when we get to the String, since we don't check if there are further nested arguments and accept the visitor, we get a null for currentTypeArgType, so we would certainly need some type of a check to make sure we don't run the logic again if we don't have further generic types.

Stack Trace: 
 java.lang.NullPointerException: Cannot invoke "com.sun.tools.javac.code.Type.cloneWithMetadata(com.sun.tools.javac.code.TypeMetadata)" because "currentTypeArgType" is null
  	at com.uber.nullaway.GenericsChecks$PreservedAnnotationTreeVisitor.visitParameterizedType(GenericsChecks.java:654)
  	at com.uber.nullaway.GenericsChecks$PreservedAnnotationTreeVisitor.visitParameterizedType(GenericsChecks.java:595)

Reply:
I fixed this by overriding the `defaultAction` method in [b9eb8fc](https://github.com/uber/NullAway/pull/806/commits/b9eb8fcaf1d24af441cb31bc7459131b9994ee83)

Reply:
That is a good idea, Thanks!